assets/src/components/detours/diversionPanel.tsx

```diff
@@ -20,57 +21,66 @@ export const DiversionPanel = ({
-      <section className="pb-3 border-bottom">
-        <h2 className="c-diversion-panel__h3">Affected route</h2>
+    <Panel.Body className="c-diversion-panel__body d-flex flex-column">
+      <Panel.Body.ScrollArea>
+        <section className="pb-3 border-bottom">
+          <h2 className="c-diversion-panel__h3">Affected route</h2>
 
-        <div className="d-flex">
-          <RoutePill
-            className="d-inline-block me-2 align-top"
-            routeName={routeName}
-          />
+          <div className="d-flex">
+            <RoutePill
+              className="d-inline-block me-2 align-top"
+              routeName={routeName}
+            />
 
-          <div className="d-inline-block">
-            <p className="my-0 c-diversion-panel__description">
-              {routeDescription}
-            </p>
-            <p className="my-0 text-muted c-diversion-panel__origin py-1">
-              {routeOrigin}
-            </p>
-            <p className="my-0 small c-diversion-panel__direction py-1">
-              {routeDirection}
-            </p>
+            <div className="d-inline-block">
+              <p className="my-0 c-diversion-panel__description">
+                {routeDescription}
+              </p>
+              <p className="my-0 text-muted c-diversion-panel__origin py-1">
+                {routeOrigin}
+              </p>
+              <p className="my-0 small c-diversion-panel__direction py-1">
+                {routeDirection}
+              </p>
+            </div>
           </div>
-        </div>
-      </section>
-
-      <section className="pb-3">
-        <h2 className="c-diversion-panel__h3">Detour Directions</h2>
-        {directions ? (
-          <ListGroup as="ol">
-            {directions.map((d) => (
-              <ListGroup.Item key={d.instruction} as="li">
-                {d.instruction}
-              </ListGroup.Item>
-            ))}
-          </ListGroup>
-        ) : (
-          <DirectionsHelpText />
-        )}
-      </section>
+        </section>
 
-      {missedStops && (
         <section className="pb-3">
-          <h2 className="c-diversion-panel__h3">Missed Stops</h2>
-          {missedStops}
+          <h2 className="c-diversion-panel__h3">Detour Directions</h2>
+          {directions ? (
+            <ListGroup as="ol">
+              {directions.map((d) => (
+                <ListGroup.Item key={d.instruction} as="li">
+                  {d.instruction}
+                </ListGroup.Item>
+              ))}
+            </ListGroup>
+          ) : (
+            <DirectionsHelpText />
+          )}
         </section>
-      )}
+
+        {missedStops && (
+          <section className="pb-3">
+            <h2 className="c-diversion-panel__h3">Missed Stops</h2>
+            {missedStops}
+          </section>
+        )}
+      </Panel.Body.ScrollArea>
+
+      <Panel.Body.Footer>
+        <Button className="flex-grow-1 m-3" onClick={onSubmit}>
```

Comment:
IIRC - the button should only appear when the detour is completely finished, right? Is that slated for a follow-up, or is that going into this PR?

Reply:
oooh great catch, thanks!
I'll do that in this PR

Reply:
How's this? fc6b00d0de6a568af012ebcae6b8e0936b824691

Reply:
Love it! Thanks!